src/App.jsx

```diff
@@ -4,14 +4,18 @@ import CategoryFilter from './components/CategoryFilter';
 import Header from './components/Header';
 import RestaurantDetailModal from './components/RestaurantDetailModal';
 import RestaurantList from './components/RestaurantList';
+import { useState } from 'react';
+import { restaurantsData } from './data/restaurantsData';
 
 function App() {
+    const [category, setCategory] = useState('전체');
```

Comment:
```suggestion
const FOOD_CATEGORY = {
  ALL: "전체",
  KOREANL "한식"
  // etc..
}

function App() {
    const [category, setCategory] = useState(FOOD_CATEGORY.ALL);
```
`전체`라는 문자열이 다른 곳에서도 사용되고 오타가 발생할 수도 있을 것 같은데 상수로 빼는건 어떨까요??🙃

Reply:
기존에 상수를 선언을 하긴 했었는데
 key랑 value가 반대로 되어있었습니다
```js
{
    전체: 'ALL',
    한식: 'korean',
    중식: 'chinese',
    일식: 'japanese',
    양식: 'western',
    아시안: 'asian',
    기타: 'etc',
}

const [category, setCategory] = useState(Object.keys(FOOD_CATEGORY).find((key) => FOOD_CATEGORY[key] === 'ALL'));
```
그래서 App.jsx에서 전체로 초기화 해줄 때 위와 같이 처리했습니다.
다만, 이렇게 해도 괜찮은지가 의문입니다.
걱정되는 사항은 가독성이 떨어지지 않을까라는 생각에 걱정을 했던 것 같습니다.
써밋은 이 부분에 대해서 어떻게 생각하시나요?🤔

Reply:
말씀해주신대로 지금 코드는 가독성이 떨어지는 문제가 있을 것 같아요!

제일 좋은 방법은 `value`가 한글로 오게 만드는 방법인데 그게 안된다고 하면 영어 ➔ 한글로 변환해주는 converter 함수를 만드는 방법도 있을 것 같아요🤔

추가적으로 useState의 초기 값 설정이 계산이 비싼 경우에는 함수로 넘겨줘서 mount 시점에만 평가하는 방식인 [게으른 초기화(lazy initialization)](https://ko.legacy.reactjs.org/docs/hooks-reference.html#lazy-initial-state)라는 것도 있어서 가볍게 한 번 보셔도 좋을 것 같습니다🙃
- as-is
```ts
// FOOD_CATEGORY의 키 값이 100,000인 경우 리렌더링이 발생할 때마다 100,000 길이의 배열을 순회
const [category, setCategory] = useState(Object.keys(FOOD_CATEGORY).find((key) => FOOD_CATEGORY[key] === 'ALL'));
```

- to-be
```ts
// 초기 값을 함수로 전달, 리렌더링이 발생해도 평가되지 않음(마운트 시점에만 useState 함수 내부에서 실행)
const [category, setCategory] = useState(() => Object.keys(FOOD_CATEGORY).find((key) => FOOD_CATEGORY[key] === 'ALL'));
```

Reply:
게으른 초기화(lazy initialization) 문서 읽어봤는데 완전 처음보는 개념이네요!
또 하나 배워갑니다!👍